### PR TITLE
Fix: Pt Polar Stat keyword

### LINKED ISSUE #3582: [BUG] pt: polar training error: KeyError: 'type'
### Bug summary

See below.

### DeePMD-kit Version

ed831c88

### TensorFlow Version

PT v2.2.0+cu121-g8ac9b20d4b0

### How did you download the software?

Built from source

### Input Files, Running Commands, Error Log, etc.

```
Traceback (most recent call last):
  File "/home/jz748/anaconda3/bin/dp", line 8, in <module>
    sys.exit(main())
  File "/home/jz748/codes/deepmd-kit/deepmd/main.py", line 807, in main
    deepmd_main(args)
  File "/home/jz748/anaconda3/lib/python3.10/site-packages/torch/distributed/elastic/multiprocessing/errors/__init__.py", line 347, in wrapper
    return f(*args, **kwargs)
  File "/home/jz748/codes/deepmd-kit/deepmd/pt/entrypoints/main.py", line 304, in main
    train(FLAGS)
  File "/home/jz748/codes/deepmd-kit/deepmd/pt/entrypoints/main.py", line 270, in train
    trainer = get_trainer(
  File "/home/jz748/codes/deepmd-kit/deepmd/pt/entrypoints/main.py", line 187, in get_trainer
    trainer = training.Trainer(
  File "/home/jz748/codes/deepmd-kit/deepmd/pt/train/training.py", line 358, in __init__
    single_model_stat(
  File "/home/jz748/codes/deepmd-kit/deepmd/pt/train/training.py", line 248, in single_model_stat
    _model.compute_or_load_stat(
  File "/home/jz748/codes/deepmd-kit/deepmd/pt/model/model/make_model.py", line 482, in compute_or_load_stat
    return self.atomic_model.compute_or_load_stat(sampled_func, stat_file_path)
  File "/home/jz748/codes/deepmd-kit/deepmd/pt/model/atomic_model/dp_atomic_model.py", line 224, in compute_or_load_stat
    self.fitting_net.compute_output_stats(wrapped_sampler, stat_file_path)
  File "/home/jz748/codes/deepmd-kit/deepmd/pt/model/task/polarizability.py", line 233, in compute_output_stats
    nframs = sampled[sys]["type"].shape[0]
KeyError: 'type'
```

### Steps to Reproduce

```sh
cd examples/water_tensor/polar
dp --pt train input_torch.json
```

### Further Information, Files, and Links

_No response_

### PR DESCRIPTION
**Pull request by anyangml**
To fix #3582, should merge after #3571.

### CODE REVIEW

**Comment by anyangml** [comment]
There seems to be another bug `KeyError: 'find_atomic_polarizability'`. 

I think the keys are filter out here. @iProzd  because the dtype for all the `find_` terms are `np.float32`
https://github.com/deepmodeling/deepmd-kit/blob/fb61efb1b16030184257f17cb4817d90568c8358/deepmd/pt/utils/stat.py#L56-L62

https://github.com/deepmodeling/deepmd-kit/blob/fb61efb1b16030184257f17cb4817d90568c8358/deepmd/utils/data.py#L666-L678

File: deepmd/pt/utils/stat.py [hunk 1]
```diff
@@ -61,12 +61,15 @@ def make_stat_input(datasets, dataloaders, nbatches):
                         if dd not in sys_stat:
                             sys_stat[dd] = []
                         sys_stat[dd].append(stat_data[dd])
-                    else:
-                        pass
+                    elif isinstance(stat_data[dd], np.float32):
+                        sys_stat[dd] = stat_data[dd]
```
Comment (iProzd) [inline deepmd/pt/utils/stat.py:65]:
```suggestion
                    elif isinstance(stat_data[dd], np.float32):
                        sys_stat[dd] = stat_data[dd]
                    else:
                        pass
```

Comment (iProzd) [reply deepmd/pt/utils/stat.py:65]: `else` situation should be kept.

Comment (iProzd) [reply deepmd/pt/utils/stat.py:65]: `else` situation should be kept.

File: deepmd/pt/utils/utils.py [hunk 2]
```diff
@@ -116,6 +116,8 @@ def dict_to_device(sample_dict):
     for key in sample_dict:
         if isinstance(sample_dict[key], list):
             sample_dict[key] = [item.to(DEVICE) for item in sample_dict[key]]
+        if isinstance(sample_dict[key], np.float32):
+            sample_dict[key] = torch.ones(1, dtype=torch.float64, device=DEVICE)
```
Comment (iProzd) [inline deepmd/pt/utils/utils.py:120]: why is this always 1 ?

Comment (anyangml) [reply deepmd/pt/utils/utils.py]: forgot the multiplication.

**Review by iProzd — APPROVED** [review]

File: deepmd/pt/utils/utils.py [hunk 3]
```diff
@@ -116,6 +116,10 @@ def dict_to_device(sample_dict):
     for key in sample_dict:
         if isinstance(sample_dict[key], list):
             sample_dict[key] = [item.to(DEVICE) for item in sample_dict[key]]
+        if isinstance(sample_dict[key], np.float32):
+            sample_dict[key] = (
+                torch.ones(1, dtype=torch.float64, device=DEVICE) * sample_dict[key]
```
Comment (njzjz) [inline deepmd/pt/utils/utils.py:121]: Why is it converted from np.float32 to torch.float64?

Comment (anyangml) [reply deepmd/pt/utils/utils.py:121]: changed back to float32

**Review by njzjz — APPROVED** [review]

**Pull request merged**